### PR TITLE
Chart streaming improvements

### LINKED ISSUE #4242: How can I stop live data streaming on a live chart, after 20 seconds for example?

### LINKED ISSUE #4258: Pause streaming chart
### Discussed in https://github.com/Megabit/Blazorise/discussions/4257

<div type='discussions-op-text'>

<sup>Originally posted by **herzplatinedurchgebrannt** November  2, 2022</sup>
Hey guys,

is there a way to pause streaming charts?

chartjs offers an pause bool in the options.
https://nagix.github.io/chartjs-plugin-streaming/latest/guide/options.html#onrefresh

Unfortunately I cannot find this option in Blazorise.

Thank you in advance!</div>

### LINKED ISSUE #4260: Suggestions for Blazorise Live Chart option (Stop, Pause, Rewind, Forward functions)
**Is your feature request related to a problem? Please describe.**
No I have suggestions (see below)

**Describe the solution you'd like**
1) The Live Chart should be stoppable any time if required. (Let say I have just 20 measuring points (each per 1 second) and I want that the X time axis movement can be stopped programatically.

2) There should be also a "Pause" function. That means I should be able to pause the chart if I want to look to the curve in detail, but the chart should collect values in the background,so that when I restart the chart I should be able to see the actual chart streaming.

3) In our language there is a saying " requesting is free of charge" :-) So I have a last request:
It would be even better If I would be able go back to the charts streaming history. (I mean rewind and forwardbuttons after I have stopped the chart.

Sorry for requesting so much :-)

### PR DESCRIPTION
**Pull request by stsrki**
Closes #4258
Closes #4260
Closes #4242

### CODE REVIEW

File: Source/Extensions/Blazorise.Charts.Streaming/ChartStreamingOptions.cs [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
         /// <summary>
         /// Frequency at which the chart is drawn on a display (frames per second). This option can be set at chart level but not at axis level. Decrease this value to save CPU power (https://github.com/nagix/chartjs-plugin-streaming#lowering-cpu-usage).
         /// </summary>
-        public int FrameRate { get; set; } = 30;
+        [JsonPropertyName( "frameRate" )]
+        [JsonIgnore( Condition = JsonIgnoreCondition.WhenWritingNull )]
+        public int? FrameRate { get; set; } = 30;
+
+        /// <summary>
+        /// If set to true, scrolling stops. Note that onRefresh callback is called even when this is set to true.
```
Comment (David-Moreira) [inline Source/Extensions/Blazorise.Charts.Streaming/ChartStreamingOptions.cs:45]: > Note that onRefresh callback is called even when this is set to true.

Anything the user should worry about?

File: Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js [hunk 2]
```diff
@@ -96,4 +94,47 @@ function getStreamingOptions(dotNetAdapter, vertical, chartOptions, streamOption
...
+
+function getRealtimeOptions(chart) {
+    if (chart && chart.options && chart.options.scales) {
+        const scales = chart.options.scales;
+
+        if (scales.x && scales.x.realtime) {
+            return scales.x.realtime;
+        }
+        else if (scales.x && scales.x.realtime) {
+            return scales.x.realtime;
```
Comment (David-Moreira) [inline Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js:107]: The else logic looks the same, is it not? Was it supposed to fallback to **y** or something?

File: Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js [hunk 3]
```diff
@@ -96,4 +94,47 @@ function getStreamingOptions(dotNetAdapter, vertical, chartOptions, streamOption
...
+export function play(canvasId, animate) {
+    const chart = getChart(canvasId);
+
+    if (chart) {
+        const realtimeOptions = getRealtimeOptions(chart);
+
+        if (realtimeOptions) {
+            realtimeOptions.pause = false;
+
+            chart.update(animate ? null : "none");
```
Comment (David-Moreira) [inline Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js:137]: I don't really get why null vs "none" depending on animate... but I imagine this is correct.

File: Source/Extensions/Blazorise.Charts.Streaming/ChartStreamingOptions.cs (hunk 1 above)
Comment (David-Moreira) [reply Source/Extensions/Blazorise.Charts.Streaming/ChartStreamingOptions.cs:45]: For example, user wants to stop a database query or api call that is being used on refresh?

**Review by David-Moreira — APPROVED** [review]
Code LGTM

PS: Do we want to add a simple Play/Pause to the docs also?  (I know its a bother to replicate demo and docs everytime haha)

File: Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js (hunk 2 above)
Comment (stsrki) [reply Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js:107]: Yep, it should be `y`.

File: Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js (hunk 3 above)
Comment (stsrki) [reply Source/Extensions/Blazorise.Charts.Streaming/wwwroot/charts.streaming.js:137]: "none" is a reserver keyword from chartjs. 🤷‍♂️

File: Source/Extensions/Blazorise.Charts.Streaming/ChartStreamingOptions.cs (hunk 1 above)
Comment (stsrki) [reply Source/Extensions/Blazorise.Charts.Streaming/ChartStreamingOptions.cs:45]: I don't know if this is wrong. We stop scrolling, but the streaming library still does its thing behind the scene. Maybe it should call refresh.

**Pull request merged**